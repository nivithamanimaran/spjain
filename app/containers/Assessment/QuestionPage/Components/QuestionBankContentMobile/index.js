import { makeStyles, Typography, Fab } from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import FilterIcon from '../../../../../assets/icons/Filter.svg';
import KenButton from '../../../../../global_components/KenButton';
import KenFooterSplitButtons from '../../../../../global_components/KenFooterSplitButtons';
import QuestionBankFiltersMobile from '../QuestionBankFiltersMobile';
const useStyles = makeStyles(theme => ({
  questionbankTypography: {
    fontWeight: 600,
    fontSize: '14px',
    color: theme.palette.KenColors.gradeSectionHeaderLabel,
  },
  filterButton: {
    position: 'absolute',
    right: theme.spacing(2),
    bottom: theme.spacing(8),
  },
}));

const QuestionBankContentMobile = props => {
  const {
    filterData,
    handleChange,
    handleExpand,
    handleClearFilter,
    getSelectedFiltersCount,
    handleQuestionBankClose,
  } = props;
  const { t } = useTranslation();
  const classes = useStyles();
  const [showFilters, setShowFilters] = useState(false);

  const handleOpenFilter = () => {
    setShowFilters(true);
  };
  const handleCloseFilter = () => {
    setShowFilters(false);
  };
  return (
    <div data-testid="questionbank-content-mobile">
      {showFilters ? (
        <QuestionBankFiltersMobile
          filterData={filterData}
          handleCloseFilter={handleCloseFilter}
          handleExpand={handleExpand}
          handleChange={handleChange}
          handleClearFilter={handleClearFilter}
          getSelectedFiltersCount={getSelectedFiltersCount}
        />
      ) : (
        <>
          <Typography className={classes.questionbankTypography}>
            {t('labels:Question_bank')}
          </Typography>
          <Fab
            color="primary"
            aria-label="filter"
            className={classes.filterButton}
            onClick={handleOpenFilter}
            data-testid="open-filters-button"
          >
            <img src={FilterIcon} />
          </Fab>
          <KenFooterSplitButtons
            primaryButtonName={'Add Selected'}
            secondaryButtonName={'Cancel'}
            secondaryAction={handleQuestionBankClose}
          />
        </>
      )}
    </div>
  );
};

export default QuestionBankContentMobile;
