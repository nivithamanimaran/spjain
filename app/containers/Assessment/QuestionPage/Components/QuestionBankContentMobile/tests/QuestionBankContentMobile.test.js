import React from 'react';
import { render, cleanup } from 'react-testing-library';
import ReactDOM from 'react-dom';
import ThemeProvider from '../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';
import QuestionBankContentMobile from '../index';
import filterData from '../../QuestionBankContent/data';
afterEach(() => {
  cleanup();
});

describe('Question bank content mobile', () => {
  const div = document.createElement('div');
  it('should render and match the snapshot', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <QuestionBankContentMobile filterData={filterData} />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankContentMobile filterData={filterData} />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('questionbank-content-mobile')).toMatchSnapshot();
  });
});
