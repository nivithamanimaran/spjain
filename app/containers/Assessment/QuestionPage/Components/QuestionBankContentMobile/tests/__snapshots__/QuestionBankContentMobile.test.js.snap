// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question bank content mobile should render and match the snapshot 1`] = `
<div
  data-testid="questionbank-content-mobile"
>
  <p
    class="MuiTypography-root makeStyles-questionbankTypography-1 MuiTypography-body1"
  >
    labels:Question_bank
  </p>
  <button
    aria-label="filter"
    class="MuiButtonBase-root MuiFab-root makeStyles-filterButton-2 MuiFab-primary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <img
        src="IMAGE_MOCK"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiGrid-root makeStyles-container-3 MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-button-4 makeStyles-secondary-6 MuiGrid-item MuiGrid-grid-xs-6"
    >
      Cancel
    </div>
    <div
      class="MuiGrid-root makeStyles-button-4 makeStyles-primary-5 MuiGrid-item MuiGrid-grid-xs-6"
    >
      Add Selected
    </div>
  </div>
</div>
`;
